tiledb/sm/consolidator/group_meta_consolidator.h

```diff
@@ -0,0 +1,136 @@
+/**
+ * @file   group_meta_consolidator.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2022 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file defines class GroupMetaConsolidator.
+ */
+
+#ifndef TILEDB_GROUP_META_CONSOLIDATOR
+#define TILEDB_GROUP_META_CONSOLIDATOR
+
+#include "tiledb/common/common.h"
+#include "tiledb/common/heap_memory.h"
+#include "tiledb/common/status.h"
+#include "tiledb/sm/consolidator/consolidator.h"
+
+using namespace tiledb::common;
+
+namespace tiledb {
+namespace sm {
+
+class StorageManager;
+
+/** Handles group metadata consolidation. */
+class GroupMetaConsolidator : public Consolidator {
+  // Declare Consolidator as friend so it can use the private constructor.
+  friend class Consolidator;
```

Comment:
Let's remove the friend declaration and also from array_meta_consolidator.h. It isn't required as I wasn't successful in making the constructors of the consolidators private.